// ngx-highlight-js/esm2020/src/highlight-js.module.mjs
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { HighlightJsDirective } from './highlight-js.directive';
import * as i0 from "@angular/core";
export class HighlightJsModule {
}
HighlightJsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: HighlightJsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
HighlightJsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: HighlightJsModule, declarations: [HighlightJsDirective], imports: [FormsModule], exports: [HighlightJsDirective] });
HighlightJsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: HighlightJsModule, imports: [[FormsModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: HighlightJsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [FormsModule],
                    declarations: [HighlightJsDirective],
                    exports: [HighlightJsDirective],
                }]
        }] });
